### ISSUE TITLE
F# exe projects fail because EntryPointAttribute attribute must be the last file

### ISSUE THREAD

**Issue by ctaggart** [AArnott/Nerdbank.GitVersioning#308]
Order is important. Any chance the generated file can be prepended to the source list instead of appended? I'm guessing that is the issue. I'm not sure where the code is yet.

```
C:\projects\froto\Compiler\Program.fs(20,5): error FS0433: A function labeled with the 'EntryPointAttribute' attribute must be the last declaration in the last file in the compilation sequence. [C:\projects\froto\Compiler\Froto.Compiler.fsproj]
```

See https://github.com/ctaggart/froto/pull/100

**Comment by AArnott** [comment]
Sounds reasonable. Thanks for reporting. In the meantime you may workaround this by turning off the source file generation:

```xml
<PropertyGroup>
  <GenerateAssemblyVersionInfo>false</GenerateAssemblyVersionInfo>
</PropertyGroup>
```

I believe the fix will be to this line:

https://github.com/AArnott/Nerdbank.GitVersioning/blob/ab61962ac8af62f9752b696d779c1324781bae8c/src/Nerdbank.GitVersioning.Tasks/build/Nerdbank.GitVersioning.targets#L166

I'm not sure if MSBuild has an `Include` alternative that prepends items instead of appending them. The fix might be to replace that line with:

```xml
    <!-- prepend the source file so we don't break F# console apps that have a "special" last source file -->
    <CompileOriginal Include="@(Compile)" />
    <Compile Remove="@(Compile)" />
    <Compile Include="$(VersionSourceFile);$(CompileOriginal)" />
```

**Comment by AArnott** [comment]
This is fixed in v2.3.138 which I just pushed to nuget.org.
